src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1693,7 +1695,7 @@ static int client_receive_message_udp(
 
         len = recv(fd, message, buflen, 0);
         if (len < 0) {
-                if (IN_SET(errno, EAGAIN, EINTR))
+                if (IN_SET(errno, EAGAIN, EINTR, ENETDOWN))
```

Comment:
a brief reference to the why for ENETDOWN would be great here too (maybe referencing the comment above)